libvast/src/system/catalog.cpp

```diff
@@ -407,9 +407,29 @@ catalog(catalog_actor::stateful_pointer<catalog_state> self,
                        atom::telemetry_v);
   }
   return {
-    [=](
-      atom::merge,
-      std::shared_ptr<std::map<uuid, partition_synopsis_ptr>>& ps) -> atom::ok {
+    [=](atom::merge,
+        std::shared_ptr<std::map<uuid, partition_synopsis_ptr>>& ps)
+      -> caf::result<atom::ok> {
+      auto unsupported_partitions = std::vector<uuid>{};
+      for (const auto& [uuid, synopsis] : *ps) {
+        auto [min, max] = version::supported_vast_version_for_partition_version(
+          synopsis->version);
+        if (max)
```

Comment:
This should probably be something like `if (max < current_partition_version)`

Reply:
Max is a VAST version, it cannot be compared with a partition version. That doesn't make sense to me.

Reply:
Ah, so if the second version is not null that already implies that the partition version is currently unsupported? This got me confused, so I'd suggest changing the names to `auto [version_introduced, version_eol] = ...` or something like that.

Reply:
I'll just make that a struct that has two members named `introduced` and `eol`, that's probably for the better.